spec/requests/api_new/users_controller_request_spec.rb

```diff
@@ -83,10 +81,24 @@
 			)
 		}
 	end
-		
 
+	context 'GET /api_new/users/current_nonprofit_object_events' do
+		let(:nonprofit_user) { create(:user_as_nonprofit_associate) }
+		let(:no_nonprofit_user) { create(:user) }
 
-  
+		it "redirects to api_new/object_events#index for the user's nonprofit when the user has one" do
+			sign_in nonprofit_user
+			get current_nonprofit_object_events_api_new_users_path
 
-end
+			np_houid = user.roles.where(host_type: 'Nonprofit').first&.host&.houid
+			assert_redirected_to controller: 'api_new/object_events', action: 'index', nonprofit_id: np_houid
+		end
+
+		it "errors when user doesn't have a nonprofit" do
+			sign_in no_nonprofit_user
+			get current_nonprofit_object_events_api_new_users_path
 
+			expect(response).to have_http_status(500)
```

Comment:
I'm guessing this isn't the best result for a user who doesn't have a nonprofit. What do u think should happen? @wwahammy